nodl_python/nodl/types.py

```diff
@@ -74,31 +101,40 @@ def __init__(
         message_type: str,
         publisher: bool = False,
         subscription: bool = False,
-        qos: QoSProfile = QoSPresetProfiles.SYSTEM_DEFAULT
+        qos: QoSProfile = QoSPresetProfiles.SYSTEM_DEFAULT.value
     ) -> None:
-        self.name = name
-        self.message_type = message_type
+        super().__init__(name=name, value_type=message_type)
         self.publisher = publisher
         self.subscription = subscription
 
         self.qos = qos
 
 
-class Node:
+class Node(NoDLData):
     """Data structure containing all interfaces a node exposes."""
 
     def __init__(
         self,
         *,
         name: str,
-        actions: Optional[List[Action]] = None,
-        parameters: Optional[List[Parameter]] = None,
-        services: Optional[List[Service]] = None,
-        topics: Optional[List[Topic]] = None
+        actions: List[Action] = [],
+        parameters: List[Parameter] = [],
+        services: List[Service] = [],
+        topics: List[Topic] = []
     ) -> None:
         self.name = name
 
-        self.actions = actions if actions else []
-        self.parameters = parameters if parameters else []
-        self.services = services if services else []
-        self.topics = topics if topics else []
+        self.actions = {action.name: action for action in actions}
+        self.parameters = {parameter.name: parameter for parameter in parameters}
+        self.services = {service.name: service for service in services}
+        self.topics = {topic.name: topic for topic in topics}
```

Comment:
Does this mean that you have you found it useful to pull specific elements out by name as opposed to just iterating through them? Where do you need that?

Reply:
A core feature will involve checking if same-named elements have the same type/parameters.